File: go/chapter10/guest_book.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	input := bufio.NewScanner(os.Stdin)

	for {
		fmt.Printf("Enter name: ")
		input.Scan()

		if input.Text() == "quit" {
			break
		}

		file, err := os.OpenFile("./guest_book.txt", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
		if err != nil {
			panic(err)
		}
		defer file.Close()

		_, err = file.WriteString(fmt.Sprintf("%s\n", input.Text()))
		if err != nil {
			panic(err)
		}

	}
}

File: javascript/chapter10/guest_book.js
import fs from "fs";
import * as readline from "readline/promises";

while (true) {
	const prompt = readline.createInterface({ input: process.stdin, output: process.stdout });
	const name = await prompt.question("Enter name: ");
	prompt.close();

	if (name.toLowerCase() == "quit") break;

	fs.appendFileSync("./guest_book.txt", name + "\n");
}
